src/lib/utils/os_utils.h

```diff
@@ -148,6 +152,10 @@ void page_allow_access(void* page);
 */
 void page_named(void* page, size_t size);
 
+#if defined(BOTAN_TARGET_OS_HAS_THREADS)
+void set_thread_name(std::thread& thread, std::string& name);
```

Comment:
Ah I guess `name` cannot be a `string_view` since we need to get the null terminated value, so I think it should be a `const std::string&`